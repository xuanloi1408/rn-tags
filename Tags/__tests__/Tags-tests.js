import React from "react";
import renderer from "react-test-renderer";
import Tags from "../../";

describe("Tags", () => {
  const noop = function() {};

  it("should render props correctly", () => {
    const tree = renderer
      .create(
        <Tags
          initialText="monkey gland sauce"
          initialTags={["tomato sauce", "mustard", "mayo"]}
          onChangeTags={noop}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("textInputProps", () => {
    const tree = renderer
      .create(
        <Tags
          initialText="monkey gland sauce"
          textInputProps={{ placeholder: "testy text" }}
          initialTags={["tomato sauce", "mustard", "mayo"]}
          onChangeTags={noop}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
