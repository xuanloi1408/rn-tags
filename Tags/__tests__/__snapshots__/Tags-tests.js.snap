// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      tomato sauce
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      mustard
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      mayo
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ccc",
          "borderRadius": 16,
          "flex": 1,
          "height": 32,
          "margin": 4,
          "minWidth": 100,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="testy text"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "flex": 1,
            "fontSize": 13,
            "height": 32,
            "margin": 0,
            "padding": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="monkey gland sauce"
    />
  </View>
</View>
`;

exports[`Tags should render props correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      tomato sauce
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      mustard
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "margin": 4,
        "opacity": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 13,
          },
          undefined,
        ]
      }
    >
      mayo
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ccc",
          "borderRadius": 16,
          "flex": 1,
          "height": 32,
          "margin": 4,
          "minWidth": 100,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "flex": 1,
            "fontSize": 13,
            "height": 32,
            "margin": 0,
            "padding": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="monkey gland sauce"
    />
  </View>
</View>
`;
